javascript/javascript1/week4/voice-assistant.js

```diff
@@ -0,0 +1,219 @@
+                }
+                else
+                {
+                    // mathematic operations
+                    let operator = wordString.slice(-2,-1);  // get the maths operator
+                    let operandA = parseInt(wordString.slice(-3,-2));
+                    let operandB = parseInt(wordString.slice(-1));
+                    let result;
+                    switch (operator[0])
+                    {
+                        case '+' :
+                        result  = operandA + operandB;
+                        console.log(`Result is: ${result}`);  
+                        break;
+
+                        case '-' :
+                        result  = operandA - operandB;
+                        console.log(`Result is: ${result}`);  
+                        break;
+                        
+                        case '*' :
+                        result  = operandA * operandB;
+                        console.log(`Result is: ${result}`);  
+                        break;
+                         
+                        case '/' :
+                        result  = operandA / operandB;
+                        console.log(`Result is: ${result}`);  
+                        break;
+
+                        case '%' :
+                        result  = operandA % operandB;
+                        console.log(`Result is: ${result}`);  
+                        break;
+
+                        default:
+                        console.log('Invalid command !');  
+                        break;  
+                    }
+                }           
+                break;
+
+            //-----------------------------------------
+            case 'Set':  
+                nameString2 = wordString.slice(1, 3); // get keyword like 'a timer'
+                nameString2 = nameString2.join(' '); // combine element to form the string
+                
+                if(nameString2 == 'a timer')
+                {
+                    let units = wordString.slice(-1); 
+                    let timervalue = wordString.slice(-2,-1);  
+                    console.log(`Timer set for ${timervalue} ${units}`);  
+                    console.log('Waiting for time out ....'); 
+                    
+                    let timeout;
+                    if((units='seconds') || (units='sec') )
+                    {
+                        timeout = timervalue *1000;  // represented a milliseconds
+                    }    
+                    else if((units = 'minutes') || (units = 'mins') )
+                    {
+                        timeout = timervalue * 1000 * 60;  // represented a milliseconds
+                    }
+
+                    setTimeout(function(){ console.log("Timer done"); }, timeout);
+
+                }
+                break;
+
+
+
+            //--------------------------------------------------------------
+            default:
+                console.log('invalid command'); 
```

Comment:
Good fallback. Potentially add `console.log('invalid command. try asking me "what day is it"')`